### PR TITLE
osd: refactor code

### PR DESCRIPTION
**Pull request by satoru-takeuchi**
**Description of your changes:**

Simplify osd management code.

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [x] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [x] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/cluster/osd/spec.go [hunk 1]
```diff
@@ -294,8 +294,8 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 	doBinaryCopyInit := true // copy rook binary in an init container?
 
 	// This property is used for both PVC and non-PVC use case
-	if osd.CVMode == "" {
-		return nil, errors.Errorf("failed to generate deployment for OSD %d. required CVMode is not specified for this OSD", osd.ID)
+	if osd.CVMode != "raw" && osd.CVMode != "lvm" {
```
Comment (travisn) [inline pkg/operator/ceph/cluster/osd/spec.go:297]: Are there any other possible CVModes than raw or lvm? If not, then I see this is equivalent to the previous condition.

File: pkg/operator/ceph/cluster/osd/spec.go [hunk 2]
```diff
@@ -356,32 +356,34 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
...
+				"--id", osdID,
+				"--fsid", c.clusterInfo.FSID,
+				"--cluster", "ceph",
+				"--setuser", "ceph",
+				"--setgroup", "ceph",
+				fmt.Sprintf("--crush-location=%s", osd.Location),
+			}
+			osd.LVBackedPV = true
+		} else {
+			doBinaryCopyInit = false
```
Comment (travisn) [inline pkg/operator/ceph/cluster/osd/spec.go:376]: How about adding a comment that this is for "raw"?

Comment (satoru-takeuchi) [reply pkg/operator/ceph/cluster/osd/spec.go:377]: Indeed,added.

File: pkg/operator/ceph/cluster/osd/spec.go (hunk 1 above)
Comment (satoru-takeuchi) [reply pkg/operator/ceph/cluster/osd/spec.go]: No. CVMode is always "raw" or "lvm". Reverted this change, thanks.

Comment (travisn) [reply pkg/operator/ceph/cluster/osd/spec.go]: That's ok if you want that condition to be clear about which types are available, I just wanted to be clear.

**Review by travisn — APPROVED** [review]

**Pull request merged**